core/src/save/level.rs

```diff
@@ -77,6 +85,74 @@ pub struct LevelVersion {
     name: String,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SuperflatGeneratorOptions {
+    pub structures: HashMap<String, nbt::Value>,
+    pub layers: Vec<SuperflatLayer>,
+    pub biome: String,
+}
+
+impl Default for SuperflatGeneratorOptions {
+    fn default() -> Self {
+        let mut default_structures = HashMap::new();
+        default_structures.insert(
+            String::from("village"),
+            nbt::Value::Compound(HashMap::new()),
+        );
+
+        // Default superflat world layers
+        Self {
+            structures: default_structures,
+            layers: vec![
+                SuperflatLayer {
+                    block: Item::Bedrock.identifier().to_string(),
+                    height: 1,
+                },
+                SuperflatLayer {
+                    block: Item::Dirt.identifier().to_string(),
+                    height: 2,
+                },
+                SuperflatLayer {
+                    block: Item::GrassBlock.identifier().to_string(),
+                    height: 1,
+                },
+            ],
+            biome: String::from("minecraft:plains"),
+        }
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct SuperflatLayer {
+    pub block: String,
```

Comment:
Should this not just be `Block`? I don't think we need to know the string value, just the block type.

Reply:
I could be wrong, but that would mean implementing Serialize and Deserialize traits manually, since `Block` doesn't map from/to the NBT string type.

Reply:
I think you can use https://serde.rs/field-attrs.html#serialize_with.

Reply:
Deserialization seems feasible, but implementing Serialization isn't feasible at the moment since there is no Block-to-identifier matching in the `block.rs` generated file. Implementing Serialize is required to implement Deserialize / use `deserialize_with` macro.

Reply:
Ah, in that case, you can just leave it as is. Perhaps add a TODO so that we can change this once those mappings are generated.